cmd/vulcan-zap/main.go

```diff
@@ -149,6 +154,39 @@ func main() {
 		logger.Debug("Waiting for spider results...")
 		time.Sleep(5 * time.Second)
 
+		logger.Debugf("Running AJAX spider %v levels deep...", opt.Depth)
+
+		client.AjaxSpider().SetOptionMaxCrawlDepth(opt.Depth)
+		resp, err = client.AjaxSpider().Scan(targetURL.String(), "", "", "")
+		if err != nil {
+			return fmt.Errorf("error executing the AJAX spider: %v", err)
+		}
+
+		for {
+			time.Sleep(10 * time.Second)
+			resp, err := client.AjaxSpider().Status()
+			if err != nil {
+				return fmt.Errorf("error getting the status of the AJAX spider: %v", err)
+			}
+
+			v, ok := resp["status"]
+			if !ok {
+				// In this case if we can not get the status let's fail.
+				return errors.New("can not retrieve the status of the AJAX spider")
+			}
+			status, ok := v.(string)
+			if !ok {
+				return errors.New("status is present in response body when calling AjaxSpider().Scatus() but it is not a string")
+			}
+
+			if status >= "running" {
```

Comment:
Just a question what are the possible statuses?

Reply:
I cannot find it documented anywhere but [according to the code](https://github.com/zaproxy/zap-extensions/blob/ec25f80911d8173b11b6cf250b8f8d7b2bd57895/addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/AjaxSpiderAPI.java#L82-L83) only two, so it should be fine.